lingocore/lib/lesson_pages.dart

```diff
@@ -0,0 +1,240 @@
+import 'package:flutter/material.dart';
+
+Color backgroundColor = Colors.white;
+
+//Question data
+class Question {
+  final String type;
+  final String questionText;
+  final List<String>? options;
+
+  Question({required this.type, required this.questionText, this.options});
+
+  factory Question.fromJson(Map<String, dynamic> json) {
+    return Question(
+      type: json['type'],
+      questionText: json['question'],
+      options:
+          json['options'] != null ? List<String>.from(json['options']) : null,
+    );
+  }
+}
+
+class QuestionWidget extends StatelessWidget {
+  final Question question;
+  final Function(String userAns) checkAns;
+  const QuestionWidget({
+    super.key,
+    required this.question,
+    required this.checkAns,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    switch (question.type) {
+      case 'true_false':
+        return buildTrueFalse();
+      case 'multiple_choice':
+        return buildMultipleChoice();
+      default:
+        return const Text('bilinmeyen soru tipi');
+    }
```

Comment:
Her soru türü için /widgets/questions.dart içinde ayrı bir class olsun.
Hatalar için material ErrorWidget kullan.
```suggestion
    switch (question.type) {
      case 'true_false':
        return TrueFalseQuestion(question);
      case 'multiple_choice':
        return MultipleChoiceQuestion(question);
      default:
        return const ErrorWidget('bilinmeyen soru tipi');
    }
```